### DISCUSSION TITLE
Can i directly use NFS paths or it has to be local paths

### DISCUSSION THREAD

**Discussion by gpradeepjain** [trapexit/mergerfs#948] [General]
Right now I am using local paths in mergerfs command. Can I directly use NFS export paths? like below

`mergerfs -o allow_other,use_ino abc.eng.com\:logs:abc-loc1.eng.com\:logs:abc-loc2.eng.com\:logs /root/merged-write-paths`

I tried the above command and the merged folder was not created as expected. Can you guide me if it is possible or mergerfs command works with local folders only?

**Comment by trapexit** [comment]
It would help if you showed what the paths actually are. Is the reason you appear to be escaping the colons is because those are part of the file path? It has been a long time since I worked on the parser for the branches but I don't know that I supported doing that. Probably should have that but all of this will be changing in the next major release anyway.

There are no such things as "local" folders and as mentioned in the docs it works with NFS. If there is a problem here it is likely what I mentioned above. You can change the mount point or use a symlink for the time being. I can look at fixing the parsing so it recognizes the escaped colons. In the future I am moving to a config file given the increased complexity and need for structure.

**Reply by gpradeepjain** [reply]
I am escaping : because delimiter defined between each paths is :.  

Below are NFS paths which I am trying to merge and /root/merged-write-paths is where it will be merged on to.
```
abc.eng.com:/logs 
abc-loc1.eng.com:/logs
abc-loc2.eng.com:/logs
```
After i execute the command , if i do an ls command i see like below and when i do cd, it says it is not even an folder

`d?????????  ? ?          ?                   ?            ? nfs-merged-write-paths
`

**Comment by trapexit** [comment]
Right. As I mentioned above... mergerfs doesn't understand '\\:' as escaping. The code would need to be modified. In the mean time you can change the names of the mounts to not include `:` or use a symlink from somewhere else pointing to those mounts (which don't have `:`). Or you could use globbing if appropriate.

**Comment by gpradeepjain** [comment]
Even if I used like `mergerfs -o allow_other,use_ino abc.eng.com:logs:abc-loc1.eng.com:logs:abc-loc2.eng.com:logs /root/merged-write-paths` same effect, i understand the reason for it. as : comes in between. Yeah for now I am mounting it on the machine and then using mergerfs on it. How do i use globbing in this scenario?

**Comment by trapexit** [comment]
Put all the mount points in the same path and use globbing as any other time.

`/mnt/nfs/*` or whatever the location is. Again... you can do that with symlinks or simply rename the mount point too.

While colon is a perfectly valid char in files in POSIX its just uncommon. At least for mount points. Colons are generally used to identify "remote" mounts by tools such as `df`. Mostly expecting to find the colon in a hostname:port sense. Of course these are remote mounts but I'm just pointing out it being uncommon in this form.

I will look to fix mergerfs but in the meantime rename the mount path or do the globbing.

That said... there some quirks (in the docs) regarding exporting mergerfs (FUSE filesystems generally) via NFS. While more configuration... using mergerfs on the client is less troublesome in some ways.

**Comment by gpradeepjain** [comment]
When you say 

> using mergerfs on the client is less troublesome in some ways.

Do you mean just merge at client-side and use for just reads?

**Comment by trapexit** [comment]
I really am not entirely sure what you're trying to accomplish. I'm just referring to a situation where you are trying to share storage over NFS (or another remote filesystem). You can use mergerfs on server A and export that to client B. Or you could export storage from server A to client B and have mergerfs on B.

**Comment by gpradeepjain** [comment]
I am not trying to expose storage over NFS. What I showed you i.e. below are my 3 NFS servers (Server+path)
abc.eng.com:/logs 
abc-loc1.eng.com:/logs
abc-loc2.eng.com:/logs

As a client, I am trying to use mergerfs to combine these NFS and write files/logs over the merged path. 

**Works**: If I mount above NFS on to local first and then use the MergerFS command on mounted folders it works like charm. I am able to see the merged folder and able to write logs as expected.

**Does not Work:** Instead of running the MergerFS command on the local mounted folder, I am trying to use it directly on NFS server paths which do not work(Description of the issue I gave above). After running the command it says the merged folder is not a folder itself.

Hope above is clear :)

**Reply by trapexit** [reply]
Yes. I know what you tried and I explained why it didn't work. What wasn't explained was the whole setup and what you were trying to accomplish at a high level. 

You can change the mount directory names or wait for me to get around to fixing it. It's just an option parsing issue as I mentioned prior.

**Reply by gpradeepjain** [reply]
Sure will go ahead with workaround and also wait for fix from your side.
